### PR TITLE
Datastore sample shows transactions, references, descendants. 

### LINKED ISSUE #1115: Datastore @Transactional integration test and sample
This should be done after ancestors work is completed because those features probably need to be demonstrated.

### PR DESCRIPTION
**Pull request by ChengyuanZhao**
fixes #1115 
sample shows transactions in 2 ways, references, descendants.

### CODE REVIEW

**Comment by ChengyuanZhao** [comment]
@meltsufin @dmitry-s This doesn't work until the `@Reference` write PR goes in. So I will ping you when that one goes in.

**Comment by ChengyuanZhao** [comment]
@meltsufin @dmitry-s the `@Reference` write is in so this sample is working now. Please take a look and give it a try!

**Comment by ChengyuanZhao** [comment]
@dmitry-s added to test! PTAL !

File: spring-cloud-gcp-samples/spring-cloud-gcp-data-datastore-sample/src/main/java/com/example/DatastoreRepositoryExample.java [hunk 1]
```diff
@@ -49,16 +53,41 @@ public CommandLineRunner commandLineRunner() {
 			System.out.println("Remove all records from 'singers' kind");
 			this.singerRepository.deleteAll();
 
-			this.singerRepository.save(new Singer("singer1", "John", "Doe", ImmutableSet.of()));
```
Comment (meltsufin) [inline spring-cloud-gcp-samples/spring-cloud-gcp-data-datastore-sample/src/main/java/com/example/DatastoreRepositoryExample.java]: This example has become one big kitchen sink, which is very difficult to follow for me, let alone for someone new to Datastore. Is there any way this big method can be broken out into smaller sub-examples that it calls?

Comment (ChengyuanZhao) [reply spring-cloud-gcp-samples/spring-cloud-gcp-data-datastore-sample/src/main/java/com/example/DatastoreRepositoryExample.java]: The issue is with running these though. There has to be just one main method. 

Do you mean just moving some code into some private methods that get called in main like "demoTransactions" ?

Comment (ChengyuanZhao) [reply spring-cloud-gcp-samples/spring-cloud-gcp-data-datastore-sample/src/main/java/com/example/DatastoreRepositoryExample.java]: @meltsufin Also I gotta say it's kind of by-design. We specifically wanted to show all these advanced features in a sample, which I think might be contrary to the nature and purpose of samples in general.

Comment (ChengyuanZhao) [reply spring-cloud-gcp-samples/spring-cloud-gcp-data-datastore-sample/src/main/java/com/example/DatastoreRepositoryExample.java]: Moved some code into some smaller methods

Comment (meltsufin) [reply spring-cloud-gcp-samples/spring-cloud-gcp-data-datastore-sample/src/main/java/com/example/DatastoreRepositoryExample.java]: Thanks! Looks better!

**Comment by ChengyuanZhao** [comment]
@meltsufin i forgot that the retrieved child entities aren't always in the same order, so I had to update the test for that. 
PTAL

**Pull request merged**